docs/tre-developers/resource-processor.md

```diff
@@ -2,76 +2,138 @@
-    ```
+## Local debugging
 
-## Local development
+To set up local debugging, first run (if you haven't done so already):
 
-To work locally in Visual Studio Code within the Dev Container you can use the `Resource Processor` debug profile to run the app.
+```cmd
+az login
+make setup-local-debugging
+```
+
+This will allowlist your local IP against Azure resources and create a Service Principal for the Resource Processor.
+
+Next, disable the existing Resource Processor from running in your deployment. The easiest way to do this is to stop the VM scale set:
+
+[![Stop RP VM scale set](../assets/rp_stop_vm_scale_set.png)](../assets/rp_stop_vm_scale_set.png)
+
+Now, go to "Run and Debug" panel in VSCode, and select Resource Processor.
+
+[![VSCode RP debugging screenshot](../assets/rp_local_debugging_vscode_screenshot.png)](../assets/rp_local_debugging_vscode_screenshot.png)
+
+!!! info
+    If you get a credential error when trying to connect to Service Bus, make sure you've authenticated in the AZ CLI first as it uses your local credentials.
+
+!!! info
+    If you get an error similar to `Environment variable 'ARM_CLIENT_ID' is not set correctly`, make sure you have ran `make setup-local-debugging`
 
-Before using this, you'll need to run `make setup-local-debugging` to whitelist your local IP against your resources and create a Service Principal for the Resource Processor to access and create Service Bus messages and deploy resources.
+You can use an API instance deployed in your environment to create deployment requests, and debug your locally running Resource Processor.
 
-Once the above is set up you can simulate receiving messages from Service Bus by going to Service Bus explorer on the portal and using a message payload for `SERVICE_BUS_RESOURCE_REQUEST_QUEUE` as follows:
+For more information on how to use API, refer to [API documentation](./api.md#using-swagger-ui).
 
-```json
-{"action": "install", "id": "a8911125-50b4-491b-9e7c-ed8ff42220f9", "name": "tre-workspace-base", "version": "0.1.0", "parameters": {"azure_location": "westeurope", "workspace_id": "20f9", "tre_id": "myfavtre", "address_space": "192.168.3.0/24"}}
+## Cloud instance
+
+On Azure Portal, find an Virtual VM scale set with a name `vmss-rp-porter-${TRE_ID}`.
+
+### Resource Processor logs in LogAnalytics
+
+To find logs in LogAnalytics, go to your resource group, then to LogAnalytics instance, which is named like `log-${TRE_ID}`.
+
+There, you can run a query like
+
+```cmd
+AppTraces 
+| where AppRoleName == "runner.py"
+| order by TimeGenerated desc 
 ```
 
-This will trigger receiving of messages, and you can freely debug the code by setting breakpoints as desired.
+### SSH-ing to the instance
+
+The processor runs in a VNET, and you cannot connect to it directly.
+To SSH to this instance, use Bastion.
+
+1. Find a keyvault with a name `kv-${TRE_ID}` in your resource group.
+1. Add yourself to the Access Policy of this keyvault with a permission to read secrets:
```

Comment:
Isn't this done for you by being the deploying user?

Reply:
I think now it is!  But it isn't on PR builds, I will add this step as optional